modules/docs/arrow-examples/src/test/kotlin/arrow/FpToTheMax.kt

```diff
@@ -63,31 +63,31 @@ object FpToTheMax {
 
     fun parseInt(s: String): Option<Int> = Try { s.toInt() }.toOption()
 
-    fun <F> MonadAndConsoleRandom<F>.checkContinue(name: String): Kind<F, Boolean> = binding {
-        putStrLn("Do you want to continue, $name?").bind()
-        val input = getStrLn().map { it.toLowerCase() }.bind()
+    fun <F> MonadAndConsoleRandom<F>.checkContinue(name: String): Kind<F, Boolean> = fx {
+        val (_) = putStrLn("Do you want to continue, $name?")
```

Comment:
Not a fan. Can we have an alias for this instead? Or keep `bind()`

Reply:
`bind` is not deprecated. This is just another supported style. The need for `bind` or `component1` destructuring only arises when a user receives a kinded value. A denoted side effect with `suspend` has nothing to destructure just function applications. This allows for programs to be defined in 3 different ways:

1. All suspended continuations denoted as effects:
```kotlin
/**
 * All side effects are marked as `suspend` functions
 */
suspend fun shouldCreateDir(file: File): Boolean =
  file.exists() && file.isDirectory

suspend fun mkdir(file: File): Boolean =
  file.createNewFile()

suspend fun tmpFile(file: File): File =
  File.createTempFile("", "", file)

suspend fun writeInto(file: File, s: String): Unit =
  file.writeText(s)

suspend fun threadName(): String =
  Thread.currentThread().name

/**
 * Direct style monadic version
 * 
 * Composisition of side effects is direct when they are encapsulated in other side effects
 * 
 * Functor, Applicative and Monad are gone from the syntax. 
 * The entire computation can't compile in a pure environment unless in the pressence of a suspend
 * capable data type.
 */
suspend fun programDirectStyle(): Unit {
  val file = File.createTempFile("tmp", "tmp")
  if (shouldCreateDir(file)) mkdir(file)
  val fh = tmpFile(file)
  writeInto(fh, threadName())
}
```

Or with wrapped pure values that encapsulate effects : 
```kotlin
/**
 * Direct style monadic version
 * 
 * Composition of side effects in `fx` blocks requires explicit `effect` denotation.
 *
 * `effect` is a compiler proof that raw suspended function are accepted in our monad continuation
 * and we have continuations to handle those unlike the raw Kotlin or Scala environment where suspended functions can't compile.
 *
 * It's implemented as:
 *
 * suspend fun <A> effect(f: suspend () -> A): A = f()
 * 
 */
fun programMonadic(): IO<Unit> = fx { // presence of an IO continuation
  effect { // denotes a block as effectful otherwise this code would not have compiled inside `fx`
    val file = File.createTempFile("tmp", "tmp")
    if (shouldCreateDir(file)) mkdir(file)
    val fh = tmpFile(file)
    writeInto(fh, threadName())
  }
}
```
or fully wrapped with all effects lifted into pure values individually:
```kotlin
/**
 * Kinded style monadic version with enhanced destructuring syntax
 * 
 * `f` lifts a `suspend () -> A` function into `Kind<F, A>` or `IO<A>`.
 *
 * Monads in Kotlin can be destructured non blocking like tuples thanks to the suspension system
 * The following expressions are equivalent and produce the effect in a controlled way:
 * val p : IO<Unit> = f { println("a") }
 * val (x : Unit) = f { println("a") }
 * val x: Unit = effect { println("a") }
 */
fun programMonadicKinded(): IO<Unit> = fx {
  val (file) = f { File.createTempFile("tmp", "tmp") }
  val (createDir) = f { shouldCreateDir(file) }
  if (createDir) f { mkdir(file) }
  val (fh) = f { tmpFile(file) }
  val (_) = f { writeInto(fh, threadName()) }
}
```

All 3 programs are equivalen and suspend effects in the same way. That is because `suspend -> A` and `F<A>` are the same thing in an `fx` block.

Reply:
My point is that `val (_)` for a side-effect is just meh. Can't we have an specialization of `bind()` for `Unit` that's like `sideEffect()` and can be grepped for.

Reply:
what would it look like when used?

Reply:
`putStrLn("Do you want to continue, $name?").sideEffect()`